lib/tz_world/geo_data.ex

```diff
@@ -0,0 +1,98 @@
+defmodule TzWorld.GeoData do
+  @moduledoc false
+
+  @compressed_data_file "timezones-geodata.etf.zip"
+  @etf_data_file "timezones-geodata.etf"
+  @default_data_dir "./priv"
+
+  defdelegate version, to: TzWorld
+
+  def default_data_dir do
+    @default_data_dir
+  end
+
+  def compressed_data_path do
+    Application.get_env(:tz_world, :data_dir, default_data_dir())
+    |> Path.join(@compressed_data_file)
+    |> to_charlist
+  end
+
+  def etf_data_path do
+    Application.get_env(:tz_world, :data_dir, default_data_dir())
+    |> Path.join(@etf_data_file)
+    |> to_charlist
+  end
+
+  def generate_compressed_data(source_data, version) do
+    geo_json = transform_source_data(source_data, version)
+    binary_data = :erlang.term_to_binary(geo_json)
+    :zip.zip(compressed_data_path(), [{etf_data_path(), binary_data}])
+  end
+
+  def load_compressed_data do
+    with {:ok, [{_, terms} | _rest]} <- :zip.unzip(compressed_data_path(), [:memory]) do
+      {:ok, :erlang.binary_to_term(terms)}
+    end
+  end
+
+  def transform_source_data(source_data, version) do
+    {:ok, [{_, json} | _rest]} = :zip.unzip(source_data, [:memory])
+
+    json
+    |> Jason.decode!()
+    |> Geo.JSON.decode!()
+    |> Map.get(:geometries)
+    |> Enum.map(&update_map_keys/1)
+    |> Enum.map(&calculate_bounding_box/1)
+    |> List.insert_at(0, version)
+  end
+
+  defp update_map_keys(%{properties: properties} = poly) do
+    properties =
+      Enum.map(properties, fn
+        {"tzid", v} -> {:tzid, v}
+        other -> other
```

Comment:
What could `other` be in this case?

Reply:
In the data as currently generated there isn't any other. But since we're working with valid GeoJSON it is legitimate to have other properties and therefore I think its reasonable to tolerate other properties since the only reason this function is here is to standardise on atom keys for the lookups.